SConstruct

```diff
@@ -1121,10 +1122,22 @@ if env['system_sundials'] == 'y':
         if retcode == 0:
             config_error("Failed to determine Sundials BLAS/LAPACK.")
         env['has_sundials_lapack'] = int(has_sundials_lapack.strip())
-    else:
-        # In Sundials 2.6, SUNDIALS_BLAS_LAPACK is either defined or undefined
+    elif sundials_ver < parse_version('5.5'):
+        # In Sundials 2.6-5.5, SUNDIALS_BLAS_LAPACK is either defined or undefined
         env['has_sundials_lapack'] = conf.CheckDeclaration('SUNDIALS_BLAS_LAPACK',
                 '#include "sundials/sundials_config.h"', 'C++')
+    else:
+        # In Sundials 5.5 and higher, two defines are included specific to the
+        # SUNLINSOL packages indicating whether SUNDIALS has been built with LAPACK
+        env["has_sundials_lapack"] = conf.CheckDeclaration(
+            "SUNDIALS_SUNLINSOL_LAPACKBAND",
+            '# include "sundials/sundials_config.h"',
+            "C++",
+        ) and conf.CheckDeclaration(
+            "SUNDIALS_SUNLINSOL_LAPACKDENSE",
+            '# include "sundials/sundials_config.h"',
+            "C++",
+        )
```

Comment:
Two formatting quibbles here:
* extraneous space between the `#` and `include` in each check
* I find this long `z = x and y` statement pretty hard to parse. I think it would be better to just break it up.